app/controllers/results_controller.rb

```diff
@@ -59,8 +59,10 @@ def edit
     @first_file = @files.first
     @extra_marks_points = @result.extra_marks.points
     @extra_marks_percentage = @result.extra_marks.percentage
-    @marks_map = Hash.new
-    @old_marks_map = Hash.new
+    @flexible_marks_map = Hash.new
```

Comment:
Don't make two different sets of variables. Anywhere you need to distinguish between the types in the views, just use the type of the first mark's criterion.